kloppy/infra/serializers/event/statsbomb/deserializer.py

```diff
@@ -319,10 +315,64 @@ def _parse_shot(shot_dict: Dict) -> Dict:
     body_part_qualifiers = _get_body_part_qualifiers(shot_dict)
     qualifiers.extend(body_part_qualifiers)
 
-    return {
-        "result": result,
-        "qualifiers": qualifiers,
-    }
+    return {"result": result, "qualifiers": qualifiers}
+
+
+def _parse_freeze_frame(
+    freeze_frame: List[Dict],
+    fidelity_version: int,
+    home_team: Team,
+    away_team: Team,
+    event: Event,
+    visible_area: Optional[List] = None,
+) -> Optional[Frame]:
+    players_data = {}
+    for i, freeze_frame_player in enumerate(freeze_frame):
+        if (event.team == home_team) == freeze_frame_player["teammate"]:
+            freeze_frame_team = home_team
+        else:
+            freeze_frame_team = away_team
+
+        if "player" in freeze_frame_player:
+            player = freeze_frame_team.get_player_by_id(
+                freeze_frame_player["player"]["id"]
+            )
+        else:
+            if freeze_frame_player.get("actor"):
+                player = event.player
+            elif freeze_frame_player.get("keeper"):
+                player = freeze_frame_team.get_player_by_position(
+                    position_id=1
+                )
+            else:
+                player = Player(
+                    player_id=f"T{freeze_frame_team.team_id}-E{event.event_id}-{i}",
+                    team=freeze_frame_team,
+                    jersey_no=None,
+                )
+
+        players_data[player] = PlayerData(
+            coordinates=_parse_coordinates(
+                freeze_frame_player["location"], fidelity_version
+            )
+        )
+
+    if event.player not in players_data:
+        # Shot data from event stream might not include coordinates for event player
+        players_data[event.player] = PlayerData(coordinates=event.coordinates)
+
+    return Frame(
+        frame_id=None,  # TODO: come up with a clever value here
```

Comment:
How about using the `event_uuid` value, potentially with a prefix or suffix?

Reply:
Was thinking about it. Now it's of type `int`. Maybe I should calculate it from the timestamp with a assumed fps of 25?

Reply:
Maybe `(event.period.end_timestamp + event.timestamp) * FPS` with `FPS = 25`